allocator_test.go

```diff
@@ -184,6 +184,74 @@ func TestTagsWithNonviableNodes(t *testing.T) {
 	require.Nil(t, allocation)
 }
 
+func TestMaxChurnWithIncreasedNodeCount(t *testing.T) {
+	const numRanges = 20
```

Comment:
Use smaller numbers, and change only a single variable, to make it obvious that the max churn limit should definitely find an answer (or not find an answer). Is testing two variants (increased node count/decreased node count) helpful? I think you should instead simply keep rf=1, nodes=1, and vary number of ranges + max churn limit to be either less than the number of new ranges (infeasible) or greater than it (feasible).